src/compiler/types.ts

```diff
@@ -2782,6 +2788,19 @@ namespace ts {
         ArrayLiteralOrCallOrNewExcludes = ContainsSpreadElementExpression,
     }
 
+    /* @internal */
+    export const enum NodeEmitFlags {
+        EmitEmitHelpers = 1 << 0,           // Any emit helpers should be written to this node.
+        EmitExportStar = 1 << 1,            // The export * helper should be written to this node.
+        EmitSuperHelper = 1 << 2,           // Emit the basic _super helper for async methods.
+        EmitAdvancedSuperHelper = 1 << 3,   // Emit the advanced _super helper for async methods.
+        UMDDefine = 1 << 4,                 // This node should be replaced with the UMD define helper.
+        NoLexicalEnvironment = 1 << 5,      // A new LexicalEnvironment should *not* be introduced when emitting this node.

```

Comment:
I can add examples, but in general this has to do with how we handle temporary variables and hoisted declarations (in SystemJS).

Generally, we control the entering or exiting of a lexical environment by calling `startLexicalEnvironment` and `endLexicalEnvironment` on `TransformationContext`. 

To maximize performance, we want to reduce or remove any unnecessary walk of a subtree. Due to the fact that some transformers will need to inspect every identifier (such as `let`/`const` in ES6, import and export bindings, etc.) we defer this until just before we emit the node in the pretty printer. If for any reason this just-in-time substitution needs to introduce a new temporary variable, we also need to handle this in the pretty printer.

The `NoLexicalEnvironment` flag supports a scenario in the module transformation for SystemJS, where any new temporary variables generated in the module body need to be declared in the outer module body function and not in the module `execute` callback. We apply `NoLexicalEnvironment` to the function we generate for the `execute` callback to ensure we don't start a new lexical environment.